src/sqlfluff/rules/L028.py

```diff
@@ -205,27 +227,29 @@ def _validate_one_reference(
     if single_table_references != this_ref_type:
         if single_table_references == "unqualified":
             # If this is qualified we must have a "table", "."" at least
-            fixes = [LintFix.delete(el) for el in ref.segments[:2]]
+            fixes = [LintFix.delete(el) for el in ref.segments[:2]] if fixable else None
             return LintResult(
                 anchor=ref,
                 fixes=fixes,
                 description="{} reference {!r} found in single table "
                 "select.".format(this_ref_type.capitalize(), ref.raw),
             )
 
-        fixes = [
```

Comment:
Review this bit with "Hide whitespace" enabled. (I just moved the `fixes = ` inside an `if`.)